sdk/typescript/src/signers/signer-with-provider.ts

```diff
@@ -104,6 +95,13 @@ export abstract class SignerWithProvider implements Signer {
     };
   }
 
+  /** @deprecated Instead of using `SignableTransaction`, pass a `Transaction` instance instead. */
+  async signTransaction(
+    transaction: SignableTransaction,
+  ): Promise<SignedTransaction>;
```

Comment:
shall we simply remove this? There's no way that all the apps that use the TS-SDK is going to continue working without a full revamp given the amount of breaking changes we have(e.g., transaction and object response). They would be better off migrating to the newest API in one shot